libctru/include/3ds/services/fspxi.h

```diff
@@ -0,0 +1,645 @@
+
+/**
+ * @brief Gets the NAND log
+ * @param out Buffer to output the log to.
+ * @param size Size of buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(GetNandLog, void* out, u32 size);
+
+/// Clears the SDMC log
+DEFINE_PXIFS_SERVICE_METHOD(ClearSdmcLog, void);
+
+/// Clears the NAND log
+DEFINE_PXIFS_SERVICE_METHOD(ClearNandLog, void);
+
+/**
+ * @brief Gets whether a card is inserted.
+ * @param inserted Pointer to output the insertion status to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardSlotIsInserted, bool* inserted);
+
+/**
+ * @brief Powers on the card slot.
+ * @param status Pointer to output the power status to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardSlotPowerOn, bool* status);
+
+/**
+ * @brief Powers off the card slot.
+ * @param status Pointer to output the power status to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardSlotPowerOff, bool* status);
+
+/**
+ * @brief Gets the card's power status.
+ * @param status Pointer to output the power status to.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardSlotGetCardIFPowerStatus, bool* status);
+
+/**
+ * @brief Executes a CARDNOR direct command.
+ * @param commandId ID of the command.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectCommand, u8 commandId);
+
+/**
+ * @brief Executes a CARDNOR direct command with an address.
+ * @param commandId ID of the command.
+ * @param address Address to provide.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectCommandWithAddress, u8 commandId, u32 address);
+
+/**
+ * @brief Executes a CARDNOR direct read.
+ * @param commandId ID of the command.
+ * @param size Size of the output buffer.
+ * @param output Output buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectRead, u8 commandId, u32 size, u8* output);
+
+/**
+ * @brief Executes a CARDNOR direct read with an address.
+ * @param commandId ID of the command.
+ * @param address Address to provide.
+ * @param size Size of the output buffer.
+ * @param output Output buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectReadWithAddress, u8 commandId, u32 address, u32 size, u8* output);
+
+/**
+ * @brief Executes a CARDNOR direct write.
+ * @param commandId ID of the command.
+ * @param size Size of the input buffer.
+ * @param output Input buffer.
+ * @remark Stubbed in latest firmware, since ?.?.?
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectWrite, u8 commandId, u32 size, u8* input);
+
+/**
+ * @brief Executes a CARDNOR direct write with an address.
+ * @param commandId ID of the command.
+ * @param address Address to provide.
+ * @param size Size of the input buffer.
+ * @param input Input buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectWriteWithAddress, u8 commandId, u32 address, u32 size, u8* input);
+
+/**
+ * @brief Executes a CARDNOR 4xIO direct read.
+ * @param commandId ID of the command.
+ * @param address Address to provide.
+ * @param size Size of the output buffer.
+ * @param output Output buffer.
+ */
+DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectRead_4xIO, u8 commandId, u32 address, u32 size, u8* output);
```

Comment:
```suggestion
DEFINE_PXIFS_SERVICE_METHOD(CardNorDirectRead_4xIO, u8 commandId, u32 address, u32 size, void* output);
```

Reply:
Same "should I propagate this to FS" question